src/rating/rating.ts

```diff
@@ -95,6 +95,11 @@ export class NgbRating implements ControlValueAccessor,
    */
   @Input() resettable: boolean;
 
+  /**
+   * A flag to know if rating should show when mouse enters or only on selection
+   */
+  @Input() mode: 'hover' | 'selection';
```

Comment:
It might be a stupid suggestion.
Why not naming the input `updateOn` with 2 different possible values: `hover` and `click`
Default one being `click` so that we are backward compatible.

Reply:
That's not a bad suggestion, but default should be `hover` to keep the current behaviour intact.

Reply:
My bad, you're totally right !! 👍